### ISSUE TITLE
Using activation function only in conv layer?

### ISSUE THREAD

**Issue by erko** [fabianbormann/Tensorflow_Segmentation#18]
Thanks for sharing good stuff! 
Reviewing this code, and found that in file DeconvNet.py ReLU used only once in convolution layer, not in deconvolution layer. In ref paper they are using. Is this correct? 
(For me, looks like problem arrives from calc of output score values that don't need activation function, linear op, but deconv need relu..)

**Comment by hungnguyen0606** [comment]
Can you please show me which paper uses the relu in deconvolution layer?

**Comment by erko** [comment]
This repository by F. Bormann (& mb others) is implementation of this paper: "Learning Deconvolution Network for Semantic Segmentation." https://arxiv.org/abs/1505.04366 (Hyeonwoo Noh et. al). In this paper there is many mentions of "activation of deconv layer", and like this line: _"The output of the deconvolutional layer is an enlarged and dense activation map."_
Also, I found original implementation in Caffee by authors of the paper:
https://github.com/HyeonwooNoh/DeconvNet/blob/master/model/DeconvNet/DeconvNet_inference_deploy.prototxt
They using relu after batch normalization.

**Comment by hungnguyen0606** [comment]
Thank you!
I also found the implementation details of Deconvolution Net in "Visualizing and Understanding Convolutional Networks", which states that we should use Relu fucntion after pooling layers. 
However, I don't know if we should use Relu after deconvolution layers.

**Comment by ghost** [comment]
as I understand the paper "Visualizing and Understanding Convolutional Networks" 
where is written:
[..] (ii) passing
the responses through a rectified linear function
(relu(x) = max(x, 0)); (iii) [optionally] max pooling[..]
the RELU activation is done on convolution layers; so BEFORE the optional max pooling layer.

On deconvolution first UnPooling then RELU and then deconvolution is done.

**Comment by fabianbormann** [comment]
fixed by e4d59e9
